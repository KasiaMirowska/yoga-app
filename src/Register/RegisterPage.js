import React from 'react';
import RegisterForm from './RegisterForm';

export default class RegisterPage extends React.Component {

  handleRegisterSuccess = () => {
    this.props.history.push('/login');
  }

  render() {
    return ( 
      <div className = 'current-flow' >
        <h2 className = 'title' > Register </h2> 
        <RegisterForm 
          onRegisterSuccess = {this.handleRegisterSuccess}
        /> 
      </div>
    );
  }
}
