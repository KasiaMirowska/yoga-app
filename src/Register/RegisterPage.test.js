import React from 'react';
import ReactDom from 'react-dom';
import renderer from 'react-test-renderer';
import RegisterPage from './RegisterPage';
import {BrowserRouter as BR} from 'react-router-dom';

describe('RegisterPage component', () => {
    const props = {match: {params: {path: '/register'}}};
    
    it ('renders without crashing',() => {
        const div = document.createElement('div');
        ReactDom.render(<BR><RegisterPage {...props} /></BR>, div);
        ReactDom.unmountComponentAtNode(div);
    });
    it('renders UI as expected', () => {
        const registerPage = renderer.create(<BR><RegisterPage /></BR>);
        expect(registerPage.toJSON()).toMatchSnapshot();
    });
});
