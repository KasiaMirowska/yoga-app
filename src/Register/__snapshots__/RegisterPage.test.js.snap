// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterPage component renders UI as expected 1`] = `
<div>
  <h2>
    Register
  </h2>
  <div>
    <form
      onSubmit={[Function]}
    >
      <label
        htmlFor="full_name"
      >
        User Full Name:
      </label>
      <input
        name="full_name"
        onChange={[Function]}
        type="text"
      />
      <label
        htmlFor="full_name"
      >
        User Name:
      </label>
      <input
        name="user_name"
        onChange={[Function]}
        required={true}
        type="text"
      />
      <label
        htmlFor="password"
      >
        Password:
      </label>
      <input
        name="password"
        onChange={[Function]}
        required={true}
        type="text"
      />
      <button
        type="reset"
      >
        Cancel
      </button>
      <button
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>
</div>
`;
